src/Features/Core/Portable/ValidateFormatString/AbstractValidateFormatStringDiagnosticAnalyzer.cs

```diff
@@ -343,8 +343,10 @@ private bool FormatCallWorksAtRuntime(string formatString, int numberOfPlacehold
             var formatStringWithEscapedBracketsChangedToSpaces = RemoveEscapedBrackets(formatString);
 
             var matches = s_extractPlaceholdersRegex.Matches(formatStringWithEscapedBracketsChangedToSpaces);
-            foreach (Match match in matches)
+            foreach (Match? match in matches)
             {
+                RoslynDebug.AssertNotNull(match);
```

Comment:
Why are we able to assert this? We already established well-formedness elsewhere?

Reply:
`match` is nullable because this is `IEnumerable` instead of `IEnumerable<Match>`.